### PR TITLE
Fixes react package

### PR DESCRIPTION
**Pull request by geclos**
## Describe your changes

- Fixed all rollup warnings during development
- Removed unnecessary dependencies
- Fixed type exports in client/core
- Fixed bug where react components were losing state on each new query
  run
- Added react-sample as a submodule and removed its code from monorepo

Lots of circular dependency warnings and missing external declarations, please pay attention to rollup when developing.

Screenshot 2024-05-14 at 15 36 25

### CODE REVIEW

File: packages/client/react/src/lib/visualizations/raw/charts/AreaChart.tsx [hunk 1]
```diff
@@ -15,7 +15,6 @@ type Props = Omit<EchartProps, 'options' | 'isComputing'> &
   Omit<CartesianChartProps, 'dataset'> &
   WrapperProps
 
-/* eslint-disable react/prop-types */
```
Comment (andresgutgon) [inline packages/client/react/src/lib/visualizations/raw/charts/AreaChart.tsx:18]: Did you have to do something to fix this?

File: packages/client/react/src/lib/utils.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
 import { theme as client } from '@latitude-data/client'
-const cn = client.utils.cn
+import clsx from 'clsx'
```
Comment (andresgutgon) [inline packages/client/react/src/lib/utils.ts:2]: Wait? why you import clsx here? Is already defined in `client`

Comment (geclos) [reply packages/client/react/src/lib/utils.ts:2]: typescript required it not sure why though. This was already a prod dependency of this package though

File: packages/client/react/src/lib/visualizations/raw/charts/AreaChart.tsx (hunk 1 above)
Comment (geclos) [reply packages/client/react/src/lib/visualizations/raw/charts/AreaChart.tsx:18]: added an eslint rule

File: packages/client/react/src/lib/utils.ts (hunk 2 above)
Comment (geclos) [reply packages/client/react/src/lib/utils.ts:2]: image

Comment (andresgutgon) [reply packages/client/react/src/lib/utils.ts:2]: Client already has that dependency https://github.com/latitude-dev/latitude/blob/main/packages/client/core/package.json#L35

Comment (andresgutgon) [reply packages/client/react/src/lib/utils.ts:2]: Util's version is doing more than `clsx` https://github.com/latitude-dev/latitude/blob/main/examples/sample-react/src/lib/utils.ts

Comment (andresgutgon) [reply packages/client/react/src/lib/utils.ts:2]: [Utility function to efficiently merge [Tailwind CSS](https://tailwindcss.com/) classes in JS without style conflicts.](https://github.com/dcastil/tailwind-merge)

File: packages/client/react/src/lib/visualizations/raw/charts/AreaChart.tsx (hunk 1 above)
Comment (andresgutgon) [reply packages/client/react/src/lib/visualizations/raw/charts/AreaChart.tsx:18]: Nice @csansoon was this : )

File: packages/client/react/src/lib/utils.ts (hunk 2 above)
Comment (geclos) [reply packages/client/react/src/lib/utils.ts]: ok then. I fixed both core and react packages.

File: packages/client/core/src/index.ts [hunk 3]
```diff
@@ -1,17 +1,16 @@
-import { type PieChartProps as PieChartPropsOriginal } from './theme/ui/chart/config/pie'
-import { type Props as CartesianChartPropsOriginal } from './theme/ui/chart/config/cartesian'
-import { type FunnelChartProps as FunnelChartPropsOriginal } from './theme/ui/chart/config/funnel'
-import { type ScatterChartProps as ScatterChartPropsOriginal } from './theme/ui/chart/config/scatter'
```
Comment (geclos) [inline packages/client/core/src/index.ts:4]: convoluted way of exporting types

File: packages/client/react/package.json [hunk 4]
```diff
@@ -34,7 +34,6 @@
     "@tanstack/react-query": "^5.28.14",
     "@tanstack/react-table": "^8.16.0",
     "class-variance-authority": "^0.7.0",
-    "clsx": "^2.1.0",
```
Comment (geclos) [inline packages/client/react/package.json:37]: not used

File: packages/client/react/rollup.config.mjs [hunk 5]
```diff
@@ -9,7 +9,17 @@ const EXTERNAL_DEPENDENCIES = [
...
+  '@radix-ui/react-popover',
+  '@radix-ui/react-slot',
+  '@radix-ui/react-label',
+  'react-table',
+  'lodash-es',
+  'echarts/core',
+  'echarts/renderers',
+  'echarts/components',
+  'echarts/charts',
+  'echarts/charts',
```
Comment (geclos) [inline packages/client/react/rollup.config.mjs:22]: all these showed up in rollup warnings

File: packages/client/react/src/lib/internal/shared/ChartWrapper.tsx [hunk 6]
```diff
@@ -1,5 +1,4 @@
 import React, { useState, useEffect, useRef } from 'react'
-import { Card } from '$src/lib'
```
Comment (geclos) [inline packages/client/react/src/lib/internal/shared/ChartWrapper.tsx:2]: circular dependency

File: packages/client/react/src/lib/internal/shared/useQuery.ts [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    }
+  }, [isLoading, setData, incoming])
+
+  return {
+    data,
+    isLoading,
+    error,
+    download: downloadFn,
+  }
+}
```
Comment (geclos) [inline packages/client/react/src/lib/internal/shared/useQuery.ts:30]: this ensures data persists during new query runs until new data shows up

File: packages/client/react/src/lib/visualizations/query/charts/FunnelChart.tsx [hunk 8]
```diff
@@ -19,15 +18,14 @@ function QueryFunnelChart({
 }: Props) {
   const {
     data,
-    isFetching: isLoading,
+    isLoading,
     error,
     download: downloadFn,
   } = useQuery({ queryPath, params, tanstaskQueryOptions })
-  const result = useMemo(() => data && new QueryResult(data), [data])
```
Comment (geclos) [inline packages/client/react/src/lib/visualizations/query/charts/FunnelChart.tsx:26]: lots of repeated code

File: packages/client/core/src/index.ts [hunk 9]
```diff
@@ -1,17 +1,16 @@
...
-export type FunnelChartProps = FunnelChartPropsOriginal
-export type ScatterChartProps = ScatterChartPropsOriginal
-export type { Dataset }
+export type { ExtendsUnion } from './lib/commonTypes'
+export type { PieChartProps } from './theme/ui/chart/config/pie'
+export type { Props as CartesianChartProps } from './theme/ui/chart/config/cartesian'
+export type { FunnelChartProps } from './theme/ui/chart/config/funnel'
+export type { ScatterChartProps } from './theme/ui/chart/config/scatter'
+export type { Dataset } from './theme/ui/chart/types'
+export type { ClassValue } from 'clsx'
```
Comment (andresgutgon) [inline packages/client/core/src/index.ts:13]: Why is this needed?

File: packages/client/react/src/index.ts [hunk 10]
```diff
@@ -1,8 +1,8 @@
...
-export { defineCustomElements } from '@latitude-data/webcomponents/dist/loader'
-
 export * from './data'
 export * from './lib'
 export * from './webcomponents'
+export * from '@latitude-data/embedding'
 export { default as LatitudeProvider } from './MainProvider'
+export { defineCustomElements } from '@latitude-data/webcomponents/dist/loader'
+
+export type { QueryResultPayload } from '@latitude-data/query_result'
```
Comment (andresgutgon) [inline packages/client/react/src/index.ts:8]: Did you start using the auto-sort plugin again? 😂

**Review by andresgutgon — APPROVED** [review]

File: packages/client/react/src/lib/internal/shared/useQuery.ts (hunk 7 above)
Comment (csansoon) [reply packages/client/react/src/lib/internal/shared/useQuery.ts:30]: Much better!

File: .changeset/rotten-pots-report.md [hunk 11]
```diff
@@ -0,0 +1,5 @@
+---
+"@latitude-data/react": patch
+---
+
+Fix: react charts lose state when running a new query.
```
Comment (csansoon) [inline .changeset/rotten-pots-report.md:5]: Changed the results of `useQuery` from regular objects to `QueryResult`. This should be at least a minor.

Comment (geclos) [reply .changeset/rotten-pots-report.md:5]: hmm not sure I get your point we are not changing the public api
